field/TinyLang/Field/Type.hs

```diff
@@ -0,0 +1,46 @@
+module TinyLang.Field.Type
+    ( Type(..)
+    , UniType
+    , pattern UniType
+    , pattern Bool
+    , pattern Field
+    , pattern Vector
+    ) where
+
+import           Data.Field
+import qualified TinyLang.Field.Uni         as U
+import           TinyLang.Field.Existential
+
+-- Types
+data Type uni f
+    = BuiltIn (Some (TypeIn uni f))
+    | TyFun   (Type uni f)          (Type uni f)
```

Comment:
We parameterize `Type` by `uni` in Plutus, because we actually need the extensibility. We're not going to need it in Zonk any time soon and if we ever need it, it'll be straightforward to add. So just replace `uni` with `Uni` and have a single `Type` instead of `Type` + `UniType`.

Reply:
Ok, I simplified `Type`.  I was considering simplifying `TypeIn`, but I will leave it as it is for now,